pkg/scheduler/preemption/preemption.go

```diff
@@ -116,6 +116,14 @@ func (p *Preemptor) GetTargets(wl workload.Info, assignment flavorassigner.Assig
 		}
 		return minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, true, allowBorrowingBelowPriority)
 	}
+
+	// the target queue is exhausted in all requested flavors, so preempting
+	// workloads from other cluster queues in the cohort is no help. so we
+	// only preempt workloads from the target cluster queue.
+	if isQueueExhaustedForAllRequestedFlavors(&wl, assignment, cq) {
+		return minimalPreemptions(&wl, assignment, snapshot, resPerFlv, sameQueueCandidates, true, nil)
+	}
+
 	targets := minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, false, nil)
 	if len(targets) == 0 {
```

Comment:
Are we still reaching this branch with any of the test cases?

Reply:
I'm tempted to remove it.

Reply:
Yes, we have a failing unit test in `preemption_test.go`: "preempting locally and borrowing other resources in cohort, without cohort candidates" (yes it is using `reclaimWithinCohort: LowerPriority`, but I checked that it also fails if we change `reclaimWithinCohort` to `Any`. I would suggest keeping it (at least in this PR, and probably its removal would require a feature gate). 

However, your comment prompted me to think of a cleaner code structure, so that `return minimalPreemptions(wlReq, cq, assignment, snapshot, resPerFlv, sameQueueCandidates, true, nil)` appears only once in code. PTAL.